cpp/src/join/sort_merge_join.cu

```diff
@@ -59,14 +87,82 @@ struct list_nonnull_filter {
   };
 };
 
+/**
+ * @brief Functor to check if a row is valid in an unprocessed table.
+ *
+ * Maps table row indices to boolean values based on the validity mask.
+ */
 struct unprocessed_table_mapper {
```

Comment:
Calling `unprocessed_table_mapper`  to check for validity of some bitmask is very confusing to me. There is nothing related to "table" here. Probably it is better named `is_valid_functor`?